xl2times/dd_to_csv.py

```diff
@@ -55,20 +55,19 @@ def parse_parameter_values_from_file(
             param_data = []
             # Parse values until a line with / is encountered.
             while not data[index].startswith("/") and data[index] != "":
-                words = data[index].split(" ")
+                line = data[index]
 
                 # Either "value" for a scalar, or "key value" for an array.
-                if len(words) == 1:
-                    attributes = []
-                elif len(words) == 2:
-                    attributes = words[0].split(".")
-                    attributes = [a if " " in a else a.strip("'") for a in attributes]
+                # So value is always the last word, or only token
+                split_point = line.rfind(" ")
+                if split_point == -1:
+                    # if only one word
+                    attributes, value = [], line
                 else:
-                    raise ValueError(
-                        f"Unexpected number of spaces in parameter value setting: {data[index]}"
-                    )
```

Comment:
This ValueError was getting thrown from veda-produced austimes DD files for a variable with a space in its name

Reply:
@SamRWest when you write "variable" do you mean an index of a GAMS parameter or a GAMS parameter?

Reply:
Sorry, 'variable' is the wrong terminology.

I mean the `key` part referred to in this comment:
`# Either "value" for a scalar, or "key value" for an array.`

Reply:
For example, here's a line from one of our DD files that (i think) would have triggered the ValueError, because of the spaces in `UC_non-decreasing EE penetration`:

`'UC_non-decreasing EE penetration'.RHS.'QLD'.2015.'EE2_Pri_Pub-b'.ANNUAL 1`

Reply:
I see, thanks! It should be okay not to raise `ValueError`, as long as the key is in quotes. Scalar is a numeric, so there should not be spaces in it.

Reply:
Cool, that was my assumption. 
The new code just splits the line at the _last_ space char, whereas the old code split it at _all_ spaces, then warned if it ended up with >2 tokens, which failed on my example string above.

Reply:
Btw, sometimes there is no value, e.g. from Demo 1:
```
SET COM_TMAP
/
'REG1'.'DEM'.'TPSCOA'
'REG1'.'NRG'.'COA'
/;
```
Let's say somebody does this:
```
SET COM_TMAP
/
'REG1'.'DEM'.'TPS COA'
'REG1'.'NRG'.'COA'
/;
```
What will happen?

Reply:
Ah, I can see we are handling this above when we check whether it is a parameter.

Reply:
Yep, the new code will catch this with this check, because `rfind` returns -1 if no spaces are found in the string.
```python
 if split_point == -1:
     #if only one word
     attributes, value = [], line
```